src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -259,8 +473,32 @@ void CommissionerDiscoveryController::Ok()
     ChipLogDetail(Controller, "------Via Shell Enter: controller ux ok [passcode]");
 }
 
+/// Callback for getting execution into the main chip thread
+void CallbackCommissionWithPasscode(System::Layer * aSystemLayer, void * aAppState)
+{
+    uint32_t * tmpPasscode = static_cast<uint32_t *>(aAppState);
+    GetCommissionerDiscoveryController()->InternalCommissionWithPasscode(*tmpPasscode);
+    delete tmpPasscode;
+}
+
 void CommissionerDiscoveryController::CommissionWithPasscode(uint32_t passcode)
 {
+    uint32_t * tmpPasscode = new uint32_t(passcode);
```

Comment:
check whether tmpPasscode is allocated successfully.

Reply:
reverted this code